### ISSUE TITLE
A bit confused about the blockchain part...

### ISSUE THREAD

**Issue by elpetak** [DemocracyEarth/sovereign#222]
# This is a General concern / Feature Proposal

#### :tophat: Where is the blockchain part implemented (if it is there) and why bitcoin is mentioned everywhere and not other blockchain alternatives?

I see bitcoin mentioned all over the place yet I couldn't find any bitcoin specific code in sovereign. Am I looking on the wrong places?
Also, why bitcoin? Bitcoin is an established crypto-currency but for general-purpose blockchain there are better options IMHO (Hyperledger, Ethereum, etc.)

**Comment by domi41** [comment]
Hello !
Sovereign is based on the blockchain, but not on Bitcoin itself. It is based on smart contracts upon the Ethereum blockchain. The related Solidity code should be published soon.

Bitcoin is mostly used as an example (from what I gather) of a successful decentralized service. If you see confusing examples that make you think that Sovereign is Bitcoin-based, please tell us where ! And welcome to the community !

**Comment by elpetak** [comment]
oh ok, so the blockchain code is not published yet?
If that's the case then I'll just wait.

What I found confusing is that while I was looking for the blockchain parts of the code I found some comments mentioning bitcoin, for example:
[https://github.com/DemocracyEarth/sovereign/blob/master/imports/api/transactions/transaction.js#L337](https://github.com/DemocracyEarth/sovereign/blob/master/imports/api/transactions/transaction.js#L337)
`  // TODO all transactions are for VOTE type, develop for BITCOIN or multi-currency conversion.
`

There are also mentions of the use of bitcoin micropayments for voting in the docs and papers.

Other repos under the same DemocracyEarth github org also relate to the use of bitcoin (although they are pretty old commits so maybe it's just testing stuff).

And the Transactions code kind of indicates that blockchain is not implemented yet [https://github.com/DemocracyEarth/sovereign/blob/master/imports/api/transactions/Transactions.js#L8](https://github.com/DemocracyEarth/sovereign/blob/master/imports/api/transactions/Transactions.js#L8)
```
/* NOTE: These schemas of Ticket & Transaction must store transactions in its own db.
*  (eventually a blockchain via vote-microchain TBD)
*/
```

**Comment by brunoksato** [comment]
Not have blockchain you can add a plugin blockchain to the code. 
Not sure how we can add this plugin and if API has support for that

**Comment by torinberger** [comment]
This project sounds so cool, keep it up guys

**Comment by torinberger** [comment]
wow @erotavlas85 that is one of the most interesting things I have ever read, very good point.

**Comment by elpetak** [comment]
I think the most important piece of advise is: you do not need a crypto currency to implement a blockchain based solution.
Blockchain != crypto currency

**Comment by erotavlas85** [comment]
> I think the most important piece of advise is: you do not need a crypto currency to implement a blockchain based solution.
> Blockchain != crypto currency
You are right, the blockchain is a distributed database managed by peer and cryptocurrencies are only an example of application. However, democracy earth is going to adopt Bitcoin as main cryptocurrency (take a look at roadmap [here](https://github.com/DemocracyEarth/sovereign)) and this is a big mistake since the blockchain of Bitcoin is the worst in term of technological features.

**Comment by brendanjones** [comment]
> However, democracy earth is going to adopt Bitcoin as main cryptocurrency (take a look at roadmap here) and this is a big mistake since the blockchain of Bitcoin is the worst in term of technological features.

What the fuuuuuck I hadn't noticed this! Thanks for pointing this out @erotavlas85. Democracy Earth people, you need to seriously reconsider this. I want you guys to succeed - please don't use Bitcoin. It's technologically flawed, and morally dubious due to its environmental impact through energy use.

Have a look at what https://github.com/secure-vote / https://github.com/voteflux (the same people) are doing. You guys really should work together (I don't mean merge your efforts, necessarily, just to share ideas for mutual benefit), for the sake of the world.

**Comment by 0bserve** [comment]
hi everyone, is there any news on when development on smart contracts is starting/being published?

**Comment by medied** [comment]
@0bserve it's already been started as we've been learning and exploring, but it's still at a somewhat early stage. I want to make repo public and welcome PRS very soon

**Comment by jnicolau** [comment]
Is it coded in solidity? I'd be happy to PR it when available.

**Comment by aecc** [comment]
@jnicolau it's using solidity. Feel free to contribute: https://github.com/DemocracyEarth/contracts
